text/0000-stabilize-marker-freeze.md

```diff
@@ -91,46 +89,75 @@ in read-only static memory or writeable static memory, or the optimizations that
 in code that holds an immutable reference to `T`.
 
 # Semver hazard
-`Freeze` being an auto-trait, it may leak private properties of your types to semver.
-Specifically, adding a non-`Freeze` field to a type's layout is a _major_ breaking change,
-as it removes a trait implementation from it. Removing the last non-`Freeeze` field of a type's
-layout is a _minor_ breaking change, as it adds a trait implementation to that type.
+`Freeze` being an auto-trait that encodes a low level property of the types it is implemented for, 
+it is strongly advised against to rely on external types maintaining that property unless that
+contract is explicitly stated out-of-band (through documentation, for example).
+
+Conversely, authors that consider `Freeze` to be part of a type's contract should document this
+fact explicitly.
 
 ## The ZST caveat
-While `UnsafeCell<T>` is currently `!Freeze` regardless of `T`, allowing `UnsafeCell<T>: Freeze` iff `T` is
+While `UnsafeCell<T>` is currently `!Freeze` regardless of `T`, allowing `UnsafeCell<T>: Freeze` if `T` is
 a Zero-Sized-Type is currently under consideration.
 
 Therefore, the advised way to make your types `!Freeze` regardless of their actual contents is to add a 
 [`PhantomNotFreeze`](core::marker::PhantomNotFreeze) field to it.
 
+[`PhantomData<T>`](core::marker::PhantomData) only implements `Freeze` if `T` does, making it a good way
+to conditionally remove a generic type's `Freeze` auto-impl.
+
 # Safety
 This trait is a core part of the language, it is just expressed as a trait in libcore for
 convenience. Do *not* implement it for other types.
 ```
 
 Mention could be added to `UnsafeCell` and atomics that adding one to a previously `Freeze` type without an indirection (such as a `Box`) is a SemVer hazard, as it will revoke its implementation of `Freeze`.
 
+## Fixing `core::marker::PhantomData`'s `Freeze` impl
+
+At time of writing, `core::marker::PhantomData<T>` implements `Freeze` regardless of whether or not `T` does.
+
+This is now considered a bug, with the corrected behaviour being that `core::marker::PhantomData<T>` only implements `Freeze` if `T` does.
+
+While crates that would "observe" this change exist, the current consensus is that it would only break invalid usages of that invariable bound.
+Most crates that would observe this change could replace their usage of `core::marker::PhantomData<T>` by `core::marker::PhantomData<Ptr<T>>` where `Ptr<T>`
+is a pointer-type with the relevant caracteristics regarding lifetime, `Send` and `Sync`ness.
+
+The author doesn't have the necessary knowledge to implement this change, which should still be subject to a crater run to ensure no valid use-cases were missed.
```

Comment:
The change is trivial, see https://github.com/rust-lang/rust/pull/114559. The hard part is dealing with all the ecosystem breakage.